iocore/eventsystem/ProtectedQueue.cc

```diff
@@ -116,13 +116,7 @@ ProtectedQueue::wait(ink_hrtime timeout)
 {
   // If there are no external events available, don't do a cond_timedwait.
```

Comment:
It is safe to access `EThread::might_have_event` because `EThread::lock` is locked.
The `EThread::lock` will be release when the Event Thread go into sleep status by `ink_cond_timedwait()` and it will be locked again when the Event Thread wake up.

Reply:
Yes, but that's not the subject of the comment. AFAICT from the code, it does a `cond_timedwait` if there are no external event available, not if there are not.